addons/sale/controllers/product_configurator.py

```diff
@@ -44,20 +54,26 @@ def _optional_product_items(self, product_id, pricelist, **kw):
             'product': product,
             'reference_product': product,
             'pricelist': pricelist,
+            # to_currency deprecated, get from pricelist or product
             'to_currency': to_currency,
+            # get_attribute_exclusions deprecated, use product method
             'get_attribute_exclusions': self._get_attribute_exclusions,
+            'add_qty': add_qty,
         })
 
     def _show_optional_products(self, product_id, variant_values, pricelist, handle_stock, **kw):
+        add_qty = int(kw.get('add_qty', 1))
         product = request.env['product.product'].browse(int(product_id))
-        from_currency = request.env.user.company_id.currency_id
         to_currency = pricelist.currency_id if pricelist else product.currency_id
         company = request.env['res.company'].browse(request.env.context.get('company_id')) or request.env['res.users']._get_company()
         date = request.env.context.get('date') or fields.Date.today()
-        compute_currency = lambda price: from_currency._convert(price, to_currency, company, date)
+
+        def compute_currency(price):
+            return product.currency_id._convert(price, to_currency, company, date)
 
         product = product.with_context(self._get_product_context(pricelist, **kw))
-        no_variant_attribute_values = request.env['product.template.attribute.value'].browse(variant_values).filtered(
+        combination = request.env['product.template.attribute.value'].browse(variant_values)
+        no_variant_attribute_values = combination.filtered(
```

Comment:
use _without_no_variant_attributes?

Reply:
This is the inverse filter, _without_no_variant_attributes has != 'no_variant' where here is '=='

Also this line can be removed on master anyway.